File: sock_lord/java/sock_lord.java
import java.util.Vector;
public class sock_lord
{
  public static void main(String[] args)
  {
    int sequence[] = {5, 5, 5, 9, 8, 7, 6, 5};
    ArrayList<int> ar = new ArrayList<int>();
    for(int i : sequence){ar.add(i);}
    Collections.sort(ar);
    int pairs = 0;
    int count = 0;
    int last = ar.get(0);
    for(int i : ar)
    {
      str = "";
      if(i == last){count++;}
      else{pairs += count / 2;}
    }
    pairs += count / 2;
    String str = "Number of pairs: " + pairs;
    System.out.println(str);
  }
}

File: sock_lord/python/sock_lord.py
def sock_lord(sequence):
    sequence.sort() #python uses an algorithm called Timsort which is
    # related to merge sort and insertion sort. Here is an informational link:
    # https://stackoverflow.com/questions/10948920/what-algorithm-does-pythons-sorted-use
    last = sequence[0]
    pairs = 0
    count = 0

    for x in sequence:
        if last == x:
            count += 1
        else:
            last = x
            pairs += int(count / 2)
            count = 1

    return pairs
if __name__ == "__main__":
    # execute only if run as a script, meaning this file is run from the command line
    # otherwise the above method declarations are the only script that runs - like "import sock_lord"
    sequence = [3, 3, 1, 1, 6, 2, 4, 8, 5, 3, 4, 6, 7, 8, 4, 2, 5, 7, 7]
    #sequence = [5, 5, 5, 9, 8, 7, 6, 5];
    print("Number of pairs: {}".format(sock_lord(sequence)))
